python_tests/test_python_frontend.py

```diff
@@ -86,16 +106,27 @@ def inner_fn(*args, **kwargs):
         inputs_copy = deepcopy(inputs)
         test_fn(self, fusion_func, inputs_copy, **kwargs)
 
-        with tempfile.NamedTemporaryFile() as tmp:
-            # Serialize FusionCache
-            fc = FusionCache.get()
-            fc.serialize(tmp.name)
+        # If DEBUG_SERDE is enabled, the temporary file is not deleted automatically
+        with tempfile.NamedTemporaryFile(delete=debug_serde) as tmp:
+            try:
+                # Serialize FusionCache
+                fc = FusionCache.get()
+                fc.serialize(tmp.name)
 
-            FusionCache.reset()
+                FusionCache.reset()
 
-            # Get new FusionCache because the previous one was destroyed by the reset call.
-            fc = FusionCache.get()
-            fc.deserialize(tmp.name)
+                # Get new FusionCache because the previous one was destroyed by the reset call.
+                fc = FusionCache.get()
+                fc.deserialize(tmp.name)
+            except Exception as e:
+                if not debug_serde:
+                    raise RuntimeError(
+                        "***** Use DEBUG_SERDE=true to debug serialization failure."
+                    )
+                else:
+                    raise RuntimeError(
+                        f"***** {tmp.name} contains the serialized binary for this failure."
```

Comment:
When debug_serde is on, this temp file won't even exist. I guess you wanted to print this when debugging is off.

Reply:
Good catch! It should be `tempfile.NamedTemporaryFile(delete=(not debug_serde))`